dao/src/main/scala/za/co/absa/enceladus/dao/rest/CrossHostApiCaller.scala

```diff
@@ -48,24 +49,34 @@ object CrossHostApiCaller {
       urlsRetryCount
     }) + 1
     val currentHostIndex = startWith.getOrElse(Random.nextInt(Math.max(apiBaseUrls.size, 1)))
-    new CrossHostApiCaller(apiBaseUrls.toVector, maxTryCount, currentHostIndex, optionallyRetryableExceptions)
+    new CrossHostApiCaller(
+      apiBaseUrls.toVector, maxTryCount, currentHostIndex, optionallyRetryableExceptions, retryBackoffStrategy
+    )
   }
 
   def apply(
       apiBaseUrls: Seq[String],
       urlsRetryCount: Int = DefaultUrlsRetryCount,
       startWith: Option[Int] = None,
-      optionallyRetryableExceptions: Set[OptRetryableExceptions] = Set.empty
+      optionallyRetryableExceptions: Set[OptRetryableExceptions] = Set.empty,
+      retryBackoffStrategy: Int => Unit = quadraticRandomizedRetryBackoffStrategy
```

Comment:
I might have missed that but it seems to me that `quadraticRandomizedRetryBackoffStrategy` is not used in tests at all. In such a case, wouldn't it make sense to cover it a bit, too?

Reply:
Changed signature from Int => Unit to Int => Int (now returns number of milliseconds to wait), and added tests.